### PR TITLE
[Network] Regenerate the ssl options and ssl predefined policy related methods

### LINKED ISSUE #26925: [BUG] `Azure.ResourceManager.Network GetApplicationGatewaySslPredefinedPolicy().Get()` throws a `System.ArgumentException` when parsing response JSON (ValidateResourceType)
### Library name and version

Azure.ResourceManager.Network 1.0.0-beta5

### Describe the bug

Attempting to retrieve an ssl predefined policy using `GetApplicationGatewaySslPredefinedPolicy().Get()`  throws a `System.ArgumentException`:

```
Message: 
System.ArgumentException : Invalid resource type Microsoft.Resources/resourceGroups/Microsoft.Network/default/AppGwSslPolicy20150501 expected Microsoft.Network/applicationGatewayAvailableSslOptions/predefinedPolicies

Stack Trace: 
ArmResource.ValidateResourceType(ResourceIdentifier identifier)
ArmResource.ctor(ClientContext clientContext, ResourceIdentifier id)
ArmResource.ctor(ArmResource parentOperations, ResourceIdentifier id)
ApplicationGatewaySslPredefinedPolicy.ctor(ArmResource options, ApplicationGatewaySslPredefinedPolicyData resource)
ApplicationGatewaySslPredefinedPolicy.Get(CancellationToken cancellationToken)
UnitTest1.Test1() line 26
```

### Expected behavior

Invoking `GetApplicationGatewaySslPredefinedPolicy(policyId).Get()` should return a valid `ApplicationGatewaySslPredefinedPolicy` object.

### Actual behavior

Invoking `GetApplicationGatewaySslPredefinedPolicy(policyId).Get()` throws the following exception:

```
Message: 
System.ArgumentException : Invalid resource type Microsoft.Resources/resourceGroups/Microsoft.Network/default/AppGwSslPolicy20150501 expected Microsoft.Network/applicationGatewayAvailableSslOptions/predefinedPolicies

Stack Trace: 
ArmResource.ValidateResourceType(ResourceIdentifier identifier)
ArmResource.ctor(ClientContext clientContext, ResourceIdentifier id)
ArmResource.ctor(ArmResource parentOperations, ResourceIdentifier id)
ApplicationGatewaySslPredefinedPolicy.ctor(ArmResource options, ApplicationGatewaySslPredefinedPolicyData resource)
ApplicationGatewaySslPredefinedPolicy.Get(CancellationToken cancellationToken)
UnitTest1.Test1() line 26
```

### Reproduction Steps

Add `Azure.Identity 1.5.0` and `Azure.ResourceManager.Network 1.0.0-beta.5` references to project:

```
<PackageReference Include="Azure.Identity" Version="1.5.0" />
<PackageReference Include="Azure.ResourceManager.Network" Version="1.0.0-beta.5" />
```

Attempt to retrieve predefined policy using `GetApplicationGatewaySslPredefinedPolicy()`:

```
ArmClient armClient = new(new DefaultAzureCredential(new DefaultAzureCredentialOptions
{
    ExcludeEnvironmentCredential = true,
    ExcludeManagedIdentityCredential = true,
    ExcludeSharedTokenCacheCredential = true,
    ExcludeVisualStudioCredential = true
}));

var subscriptionId = "<...>";

var policyName = ApplicationGatewaySslPolicyName.AppGwSslPolicy20150501.ToString();
var policyId = ApplicationGatewaySslPredefinedPolicy.CreateResourceIdentifier(subscriptionId, policyName);
var policy = armClient.GetApplicationGatewaySslPredefinedPolicy(policyId).Get();
```

### Environment

Using Microsoft Visual Studio Enterprise 2022 (64-bit) Version 17.0.5 and targeting .NET 6.0.101.

```
dotnet --info
.NET SDK (reflecting any global.json):
 Version:   6.0.101
 Commit:    ef49f6213a

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19042
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.101\

Host (useful for support):
  Version: 6.0.1
  Commit:  3a25a7f1cc

.NET SDKs installed:
  3.1.201 [C:\Program Files\dotnet\sdk]
  3.1.301 [C:\Program Files\dotnet\sdk]
  3.1.407 [C:\Program Files\dotnet\sdk]
  5.0.104 [C:\Program Files\dotnet\sdk]
  5.0.404 [C:\Program Files\dotnet\sdk]
  6.0.101 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.26 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.26 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.22 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.26 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.22 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.13 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.22 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.13 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

### PR DESCRIPTION
**Pull request by Yao725**
Fix #26925

This PR includes:
1. Re-generate the methods `ApplicationGateways_ListAvailableSslOptions`, `ApplicationGateways_ListAvailableSslPredefinedPolicies` and `ApplicationGateways_GetSslPredefinedPolicy`(Considering that these three methods are just `Get` method, I marked them as non-resource).
2. Polish the new added APIs.
3. Add tests for the new methods.

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs [hunk 1]
```diff
@@ -1877,6 +1877,10 @@ public static partial class NetworkExtensions
...
         public static Azure.Pageable<Azure.ResourceManager.Network.Models.AvailableServiceAlias> GetAvailableServiceAliasesByResourceGroup(this Azure.ResourceManager.Resources.ResourceGroupResource resourceGroupResource, Azure.Core.AzureLocation location, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public static Azure.AsyncPageable<Azure.ResourceManager.Network.Models.AvailableServiceAlias> GetAvailableServiceAliasesByResourceGroupAsync(this Azure.ResourceManager.Resources.ResourceGroupResource resourceGroupResource, Azure.Core.AzureLocation location, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static Azure.Response<Azure.ResourceManager.Network.Models.ApplicationGatewayAvailableSslOptions> GetAvailableSslOptionsApplicationGateway(this Azure.ResourceManager.Resources.SubscriptionResource subscriptionResource, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (ArcturusZhang) [inline sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs:1880]: I believe `GetApplicationGatewayAvailableSslOptions` makes more sense?
And should we rename this model to make sure it does not end with `Options`?

File: sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs [hunk 2]
```diff
@@ -1877,6 +1877,10 @@ public static partial class NetworkExtensions
...
+        public static System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.Network.Models.ApplicationGatewayAvailableSslOptions>> GetAvailableSslOptionsApplicationGatewayAsync(this Azure.ResourceManager.Resources.SubscriptionResource subscriptionResource, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static Azure.Pageable<Azure.ResourceManager.Network.Models.ApplicationGatewaySslPredefinedPolicy> GetAvailableSslPredefinedPoliciesApplicationGateways(this Azure.ResourceManager.Resources.SubscriptionResource subscriptionResource, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (ArcturusZhang) [inline sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs:1882]: Same
```suggestion
        public static Azure.Pageable<Azure.ResourceManager.Network.Models.ApplicationGatewaySslPredefinedPolicy> GetApplicationGatewayAvailableSslPredefinedPolicies(this Azure.ResourceManager.Resources.SubscriptionResource subscriptionResource, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```

File: sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs [hunk 3]
```diff
@@ -2103,6 +2107,8 @@ public static partial class NetworkExtensions
...
         public static Azure.Response<Azure.ResourceManager.Network.Models.ServiceTagsListResult> GetServiceTag(this Azure.ResourceManager.Resources.SubscriptionResource subscriptionResource, Azure.Core.AzureLocation location, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public static System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.Network.Models.ServiceTagsListResult>> GetServiceTagAsync(this Azure.ResourceManager.Resources.SubscriptionResource subscriptionResource, Azure.Core.AzureLocation location, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static Azure.Response<Azure.ResourceManager.Network.Models.ApplicationGatewaySslPredefinedPolicy> GetSslPredefinedPolicyApplicationGateway(this Azure.ResourceManager.Resources.SubscriptionResource subscriptionResource, string predefinedPolicyName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (ArcturusZhang) [inline sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs:2110]:
```suggestion
        public static Azure.Response<Azure.ResourceManager.Network.Models.ApplicationGatewaySslPredefinedPolicy> GetApplicationGatewaySslPredefinedPolicy(this Azure.ResourceManager.Resources.SubscriptionResource subscriptionResource, string predefinedPolicyName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```

File: sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs (hunk 1 above)
Comment (Yao725) [reply sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs]: fixed and rename the model to `ApplicationGatewayAvailableSslOptionsInfo`.

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**